api/envoy/api/v2/lds.proto

```diff
@@ -194,4 +195,13 @@ message Listener {
 
   // Specifies the intended direction of the traffic relative to the local Envoy.
   core.TrafficDirection traffic_direction = 16;
+
+  // If the protocol in the listener socket address in :ref:`protocol
+  // <envoy_api_field_core.SocketAddress.protocol>` is :ref:'UDP
+  // <envoy_api_field_core.Protocol.UDP>`, this field specifies the actual udp listener to create,
+  // i.e. :ref:`udp_listener_name
+  // <envoy_api_field_listener.UdpListenerConfig.udp_listener_name>` =
+  // "raw_udp_listener" for creating an ActiveUdpListener. If not present, treat it as
+  // "raw_udp_listener".
+  listener.UdpListenerConfig udp_listener_config = 18;
```

Comment:
Will UDP listeners have their own FilterChains? Do we want to aim for a relatively similar and consistent API between TCP and UDP listener filter chain matchers?

Reply:
Filter chain design is here: https://docs.google.com/document/d/1-CfWo0t823nT9WA8wZQk0qNLolgb1jh_4HGbNbLRbZk/edit#. QuicFilterChain should have the same protobuf, and the same  matchers. The QuicFilterChain implementation will be different from FilterChain because Quic doesn't have transport socket.

Reply:
@htuch if you are keeping notes this is one of the things we need to clean up in v3. We need to figure out how to make some of the listener stuff polymorphic.